src/WorkerChannel.ts

```diff
@@ -25,12 +26,17 @@ interface IWorkerChannel {
   invocationRequest(requestId: string, msg: rpc.InvocationRequest): void;
   invocationCancel(requestId: string, msg: rpc.InvocationCancel): void;
   functionEnvironmentReloadRequest(requestId: string, msg: rpc.IFunctionEnvironmentReloadRequest): void;
+  invocationRequestBefore?: (context: Context, userFn: Function) => Function;
+  invocationRequestAfter?: Function;
 }
 
 /**
  * Initializes handlers for incoming gRPC messages on the client
  */
 export class WorkerChannel implements IWorkerChannel {
+  public invocationRequestBefore?: (context: Context, userFn: Function) => Function;
+  public invocationRequestAfter?: Function;

```

Comment:
Curious as to why `context` isn't exposed in `After` as well.

Reply:
Agree. It seems like it would be required to correlate internal logging with the trace context